File: scripts/create-pocketbase-schema.py
#!/usr/bin/env python3
"""
Create Pocketbase collections from our Drizzle schema
Run: python3 scripts/create-pocketbase-schema.py
"""

import requests
import json

BASE_URL = "http://127.0.0.1:8090"
ADMIN_EMAIL = "admin@localhost.com"
ADMIN_PASSWORD = "test1234567"

# Authenticate (admin auth endpoint)
auth_response = requests.post(
    f"{BASE_URL}/api/admins/auth-via-email",
    json={"email": ADMIN_EMAIL, "password": ADMIN_PASSWORD}
)
auth_response.raise_for_status()
token = auth_response.json()["token"]

headers = {"Authorization": token}

print("✓ Authenticated\n")
print("Creating collections...\n")

# Define all collections
collections = [
    {
        "name": "tags",
        "type": "base",
        "fields": [
            {"name": "description", "type": "text", "required": False},
            {"name": "category", "type": "text", "required": False},
            {"name": "status", "type": "text", "required": True, "max": 50}
        ]
    },
    {
        "name": "organizations",
        "type": "base",
        "fields": [
            {"name": "name", "type": "text", "required": True},
            {"name": "description", "type": "text", "required": False},
            {"name": "website", "type": "url", "required": False},
            {"name": "logo", "type": "text", "required": False},
            {"name": "status", "type": "text", "required": True, "max": 50}
        ]
    },
    {
        "name": "teams",
        "type": "base",
        "fields": [
            {"name": "name", "type": "text", "required": True},
            {"name": "description", "type": "text", "required": False},
            {"name": "status", "type": "text", "required": True, "max": 50}
        ]
    },
    {
        "name": "technologies",
        "type": "base",
        "fields": [
            {"name": "name", "type": "text", "required": True},
            {"name": "description", "type": "text", "required": False},
            {"name": "website", "type": "url", "required": False},
            {"name": "logo", "type": "text", "required": False},
            {"name": "category", "type": "text", "required": False},
            {"name": "type", "type": "text", "required": False},
            {"name": "status", "type": "text", "required": True, "max": 50}
        ]
    },
    {
        "name": "standards",
        "type": "base",
        "fields": [
            {"name": "name", "type": "text", "required": True},
            {"name": "description", "type": "text", "required": False},
            {"name": "website", "type": "url", "required": False},
            {"name": "type", "type": "text", "required": False},
            {"name": "category", "type": "text", "required": False},
            {"name": "vendor", "type": "text", "required": False},
            {"name": "version", "type": "text", "required": False},
            {"name": "logo", "type": "text", "required": False},
            {"name": "status", "type": "text", "required": True, "max": 50}
        ]
    },
    {
        "name": "profiles",
        "type": "base",
        "fields": [
            {"name": "name", "type": "text", "required": True},
            {"name": "version", "type": "text", "required": False},
            {"name": "platform", "type": "text", "required": False},
            {"name": "framework", "type": "text", "required": False},
            {"name": "vendor", "type": "text", "required": False},
            {"name": "github", "type": "url", "required": False},
            {"name": "details", "type": "text", "required": False},
            {"name": "standard_version", "type": "text", "required": False},
            {"name": "short_description", "type": "text", "required": False},
            {"name": "requirements", "type": "text", "required": False},
            {"name": "category", "type": "text", "required": False},
            {"name": "status", "type": "text", "required": True, "max": 50}
        ]
    },
    {
        "name": "hardening_profiles",
        "type": "base",
        "fields": [
            {"name": "name", "type": "text", "required": True},
            {"name": "version", "type": "text", "required": False},
            {"name": "platform", "type": "text", "required": False},
            {"name": "type", "type": "text", "required": False},
            {"name": "vendor", "type": "text", "required": False},
            {"name": "github", "type": "url", "required": False},
            {"name": "details", "type": "text", "required": False},
            {"name": "standard_version", "type": "text", "required": False},
            {"name": "short_description", "type": "text", "required": False},
            {"name": "requirements", "type": "text", "required": False},
            {"name": "category", "type": "text", "required": False},
            {"name": "difficulty", "type": "text", "required": False},
            {"name": "status", "type": "text", "required": True, "max": 50}
        ]
    },
    {
        "name": "tools",
        "type": "base",
        "fields": [
            {"name": "name", "type": "text", "required": True},
            {"name": "version", "type": "text", "required": False},
            {"name": "description", "type": "text", "required": False},
            {"name": "website", "type": "url", "required": False},
            {"name": "logo", "type": "text", "required": False},
            {"name": "github", "type": "url", "required": False},
            {"name": "category", "type": "text", "required": False},
            {"name": "status", "type": "text", "required": True, "max": 50}
        ]
    },
    {
        "name": "capabilities",
        "type": "base",
        "fields": [
            {"name": "name", "type": "text", "required": True},
            {"name": "description", "type": "text", "required": False},
            {"name": "category", "type": "text", "required": False},
            {"name": "status", "type": "text", "required": True, "max": 50}
        ]
    }
]

# Create each collection
for collection in collections:
    print(f"Creating {collection['name']}...")
    response = requests.post(
        f"{BASE_URL}/api/collections",
        headers=headers,
        json=collection
    )
    if response.status_code != 200:
        print(f"  ❌ Error: {response.text}")
    else:
        print(f"  ✓ {collection['name']} created")

print("\n✅ All collections created!")
print("\nRefresh your admin UI at http://localhost:8090/_/")
print("\nNote: Foreign key relations need to be added manually in the UI:")
print("  - teams.organization → organizations")
print("  - profiles.technology → technologies")
print("  - profiles.organization → organizations")
print("  - profiles.team → teams")
print("  - profiles.standard → standards")
print("  - hardening_profiles.technology → technologies")
print("  - hardening_profiles.organization → organizations")
print("  - hardening_profiles.team → teams")
print("  - hardening_profiles.standard → standards")
print("  - tools.technology → technologies")
print("  - tools.organization → organizations")

File: scripts/create-pocketbase-schema.ts
#!/usr/bin/env tsx
/**
 * Create Pocketbase collections from our Drizzle schema
 * Run: pnpm tsx scripts/create-pocketbase-schema.ts
 */

import PocketBase from 'pocketbase'

const pb = new PocketBase('http://127.0.0.1:8090')

const adminEmail = 'admin@localhost.com'
const adminPassword = 'test1234567'

async function createCollections() {
  try {
    // Login as admin
    console.log('Authenticating as admin...')
    await pb.admins.authWithPassword(adminEmail, adminPassword)
    console.log('✓ Authenticated\n')

    console.log('PHASE 1: Creating base collections without relations...\n')

    // 1. Tags
    console.log('Creating tags...')
    await pb.collections.create({
      name: 'tags',
      type: 'base',
      fields: [
        { name: 'description', type: 'text', required: false },
        { name: 'category', type: 'text', required: false },
        { name: 'status', type: 'text', required: true, max: 50 },
      ],
    })

    // 2. Organizations
    console.log('Creating organizations...')
    await pb.collections.create({
      name: 'organizations',
      type: 'base',
      schema: [
        { name: 'name', type: 'text', required: true },
        { name: 'description', type: 'text', required: false },
        { name: 'website', type: 'url', required: false },
        { name: 'logo', type: 'text', required: false },
        { name: 'status', type: 'text', required: true, options: { max: 50 } },
      ],
    })

    // 3. Teams (without relation initially)
    console.log('Creating teams...')
    await pb.collections.create({
      name: 'teams',
      type: 'base',
      schema: [
        { name: 'name', type: 'text', required: true },
        { name: 'description', type: 'text', required: false },
        { name: 'status', type: 'text', required: true, options: { max: 50 } },
      ],
    })

    // 4. Technologies
    console.log('Creating technologies...')
    await pb.collections.create({
      name: 'technologies',
      type: 'base',
      schema: [
        { name: 'name', type: 'text', required: true },
        { name: 'description', type: 'text', required: false },
        { name: 'website', type: 'url', required: false },
        { name: 'logo', type: 'text', required: false },
        { name: 'category', type: 'text', required: false },
        { name: 'type', type: 'text', required: false },
        { name: 'status', type: 'text', required: true, options: { max: 50 } },
      ],
    })

    // 5. Standards
    console.log('Creating standards...')
    await pb.collections.create({
      name: 'standards',
      type: 'base',
      schema: [
        { name: 'name', type: 'text', required: true },
        { name: 'description', type: 'text', required: false },
        { name: 'website', type: 'url', required: false },
        { name: 'type', type: 'text', required: false },
        { name: 'category', type: 'text', required: false },
        { name: 'vendor', type: 'text', required: false },
        { name: 'version', type: 'text', required: false },
        { name: 'logo', type: 'text', required: false },
        { name: 'status', type: 'text', required: true, options: { max: 50 } },
      ],
    })

    // 6. Validation Profiles (without relations)
    console.log('Creating profiles...')
    await pb.collections.create({
      name: 'profiles',
      type: 'base',
      schema: [
        { name: 'name', type: 'text', required: true },
        { name: 'version', type: 'text', required: false },
        { name: 'platform', type: 'text', required: false },
        { name: 'framework', type: 'text', required: false },
        { name: 'vendor', type: 'text', required: false },
        { name: 'github', type: 'url', required: false },
        { name: 'details', type: 'text', required: false },
        { name: 'standard_version', type: 'text', required: false },
        { name: 'short_description', type: 'text', required: false },
        { name: 'requirements', type: 'text', required: false },
        { name: 'category', type: 'text', required: false },
        { name: 'status', type: 'text', required: true, options: { max: 50 } },
      ],
    })

    // 7. Hardening Profiles (without relations)
    console.log('Creating hardening_profiles...')
    await pb.collections.create({
      name: 'hardening_profiles',
      type: 'base',
      schema: [
        { name: 'name', type: 'text', required: true },
        { name: 'version', type: 'text', required: false },
        { name: 'platform', type: 'text', required: false },
        { name: 'type', type: 'text', required: false },
        { name: 'vendor', type: 'text', required: false },
        { name: 'github', type: 'url', required: false },
        { name: 'details', type: 'text', required: false },
        { name: 'standard_version', type: 'text', required: false },
        { name: 'short_description', type: 'text', required: false },
        { name: 'requirements', type: 'text', required: false },
        { name: 'category', type: 'text', required: false },
        { name: 'difficulty', type: 'text', required: false },
        { name: 'status', type: 'text', required: true, options: { max: 50 } },
      ],
    })

    // 8. Tools (without relations)
    console.log('Creating tools...')
    await pb.collections.create({
      name: 'tools',
      type: 'base',
      schema: [
        { name: 'name', type: 'text', required: true },
        { name: 'version', type: 'text', required: false },
        { name: 'description', type: 'text', required: false },
        { name: 'website', type: 'url', required: false },
        { name: 'logo', type: 'text', required: false },
        { name: 'github', type: 'url', required: false },
        { name: 'category', type: 'text', required: false },
        { name: 'status', type: 'text', required: true, options: { max: 50 } },
      ],
    })

    // 9. Capabilities
    console.log('Creating capabilities...')
    await pb.collections.create({
      name: 'capabilities',
      type: 'base',
      schema: [
        { name: 'name', type: 'text', required: true },
        { name: 'description', type: 'text', required: false },
        { name: 'category', type: 'text', required: false },
        { name: 'status', type: 'text', required: true, options: { max: 50 } },
      ],
    })

    console.log('\n✅ Phase 1 complete! All base collections created.')
    console.log('\nNote: Foreign key relations need to be added manually in the admin UI.')
    console.log('The following relations are needed:')
    console.log('  - teams.organization → organizations')
    console.log('  - profiles.technology → technologies')
    console.log('  - profiles.organization → organizations')
    console.log('  - profiles.team → teams')
    console.log('  - profiles.standard → standards')
    console.log('  - hardening_profiles.technology → technologies')
    console.log('  - hardening_profiles.organization → organizations')
    console.log('  - hardening_profiles.team → teams')
    console.log('  - hardening_profiles.standard → standards')
    console.log('  - tools.technology → technologies')
    console.log('  - tools.organization → organizations')
    console.log('\nRefresh your admin UI at http://localhost:8090/_/')
  }
  catch (error) {
    console.error('❌ Error:', error)
    process.exit(1)
  }
}

createCollections()
